src/js/abstract/base-component.js

```diff
@@ -176,11 +184,29 @@ export default class BaseComponent extends HTMLElement {
       lifecycleLogger(this.logLifecycle)(`+++ attributeChangedCallback -> ${this.nodeName}#${this._id} | ${name} from ${oldValue} to ${newValue}\n`);
     }
 
+    // only update the value if it has actually changed
+    // and only re-render if it has changed
+    if (!this.shouldComponentUpdate(newValue, oldValue)) {
+      return;
+    }
+
     const key = camelize(name);
 
     this[key] = toProp(newValue);
   }
 
+  /**
+   * Check if a re-render is really necessary.
+   * Basic check does a shallow comparison.
+   *
+   * @param {*} newValue - the new value of an attribute.
+   * @param {*} oldValue - the existing value of an attribute.
+   * @returns {Boolean} - Returns `true` if attributes have changed, else `false`.
+   */
+  shouldComponentUpdate(newValue, oldValue) {
```

Comment:
@LucaMele 
I think it's better to rename it to `shouldUpdateCallback()` to distinguish it from React's `shouldComponentUpdate()` and to be consistent with Custom-Element's naming of lifecycle methods 🤔